setup.py

```diff
@@ -78,156 +87,135 @@ def find_javac(possible_homes):
-    # It is needed to set the correct lib path, found in the jre_home, eg.
-    # <jre_home>/lib/<cpu>/.
-    machine2cpu = {
-        "i686" : "i386",
-        "x86_64" : "amd64",
-        "armv7l" : "arm"
+    # This dictionary converts values from platform.machine()
+    # to a "cpu" string. It is needed to set the correct lib path,
+    # found in the JRE_HOME, e.g.: <JRE_HOME>/lib/<cpu>/.
+    MACHINE2CPU = {
+        "i686": "i386",
+        "x86_64": "amd64",
+        "armv7l": "arm"
     }
-    if machine() in machine2cpu.keys():
-        cpu = machine2cpu[machine()]
+    if machine() in MACHINE2CPU.keys():
+        CPU = MACHINE2CPU[machine()]
     else:
-        print("WARNING: Not able to assign machine() = %s to a cpu value!" %(machine()))
+        print(
+            "WARNING: Not able to assign machine()"
+            " = %s to a cpu value!" % machine()
+        )
         print("         Using cpu = 'i386' instead!")
-        cpu = 'i386'
+        CPU = 'i386'
 
-    if platform == 'win32':
-        incl_dir = join(jdk_home, 'include', 'win32')
-        libraries = ['jvm']
+    if PLATFORM == 'win32':
+        INCL_DIR = join(JDK_HOME, 'include', 'win32')
+        LIBRARIES = ['jvm']
     else:
-        incl_dir = join(jdk_home, 'include', 'linux')
-        lib_location = 'jre/lib/{}/server/libjvm.so'.format(cpu)
+        INCL_DIR = join(JDK_HOME, 'include', 'linux')
+        LIB_LOCATION = 'jre/lib/{}/server/libjvm.so'.format(CPU)
 
-    include_dirs = [
-        join(jdk_home, 'include'),
-        incl_dir
+    INCLUDE_DIRS = [
+        join(JDK_HOME, 'include'),
+        INCL_DIR
     ]
 
-    if platform == 'win32':
-        library_dirs = [
-            join(jdk_home, 'lib'),
-            join(jre_home, 'bin', 'server')
+    if PLATFORM == 'win32':
+        LIBRARY_DIRS = [
+            join(JDK_HOME, 'lib'),
+            join(JRE_HOME, 'bin', 'server')
         ]
 
-    compile_native_invocation_handler(jdk_home, jre_home)
+    compile_native_invocation_handler(JDK_HOME, JRE_HOME)
 
 # generate the config.pxi
 with open(join(dirname(__file__), 'jnius', 'config.pxi'), 'w') as fd:
-    fd.write('DEF JNIUS_PLATFORM = {0!r}\n\n'.format(platform))
+    fd.write('DEF JNIUS_PLATFORM = {0!r}\n\n'.format(PLATFORM))
     if PY3:
         fd.write('DEF JNIUS_PYTHON3 = True\n\n')
     else:
         fd.write('DEF JNIUS_PYTHON3 = False\n\n')
-    if lib_location is not None:
-        fd.write('DEF JNIUS_LIB_SUFFIX = {0!r}\n\n'.format(lib_location))
-
-with open(join('jnius', '__init__.py')) as fd:
-    versionline = [x for x in fd.readlines() if x.startswith('__version__')]
-    version = versionline[0].split("'")[-2]
+    if LIB_LOCATION is not None:
+        fd.write('DEF JNIUS_LIB_SUFFIX = {0!r}\n\n'.format(LIB_LOCATION))
 
 # create the extension
 setup(
-    name='pyjnius',
-    version=version,
     cmdclass={'build_ext': build_ext},
-    packages=['jnius'],
```

Comment:
Sdist means we don't create any wheels, we just release it the same way you can install it from GitHub:

    pip install https://github.com/kivy/pyjnius/zipball/master

and separating the keys means I can safely create the source archive and upload it to PyPI with all the classifiers and other metadata in tact. Also it means a GNU/Linux user can install it from PyPI the same way eg. Cython is installed on RPi - compilation is triggered.

It can compile the normal way and even via pip if you run the command from above.

Log:

https://travis-ci.org/kivy/pyjnius/jobs/441130344#L624